webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +121,18 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            // only if ?next=/... exists on the current URL
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                // Make sure it's on this server
+                                document.location.pathname = next;
+                            } else {
```

Comment:
@willkg Annoyingly, all the previous discussions disappeared when I made this new commit. 

First of all, `URL` doesn't work for us. If you do `new URL('/foo/', document.location.href).href` it nicely takes the base from that second parameter and puts `/foo/` as the pathname. 
But if you do `new URL('http://www.example.com/something', document.location.href).href` the output becomes `http://www.example.com/something`. 

Also, this `new URL(...).searchParams` I don't see anywhere. Certainly not in Firefox. Perhaps we're talking about different things. 

Lastly, one important change I made here is that I switched to using `document.location.PATHNAME = next`. So if someone tries to be funny and tricks me to go to `https://crash-stats.m.c/login/?next=http://pr0n.net/spam` it will attempt to redirect to `https://crash-stats.m.c/http:/pr0n.net/spam` which will simply 404.